src/lib.rs

```diff
@@ -1404,6 +1404,11 @@ pub enum Expression {
     CallResult(Handle<Function>),
     /// Result of an atomic operation.
     AtomicResult { ty: Handle<Type>, comparison: bool },
+    /// Result of a [`Statement::WorkgroupUniformLoad`]
+    ///
+    /// This expression is uniform by definition
+    /// (The pointer here is stored for type resolution only)
+    WorkgroupUniformLoadResult { pointer: Handle<Expression> },
```

Comment:
```suggestion
    WorkgroupUniformLoadResult { ty: Handle<Type> },
```

I think we should keep this definition consistent with `AtomicResult`.